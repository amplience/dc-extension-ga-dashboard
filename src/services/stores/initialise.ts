import {
  contentItemFilter,
  editionFilter,
  slotFilter,
} from '../../stores/ga-query-filters';
import {
  breakdown,
  contentItemIdMapping,
  editionIdMapping,
  gaApiKey,
  gaClientEmail,
  gaClientId,
  gaViewId,
  slotIdMapping,
} from '../../stores/google-analytics';
import { currencyCode, locale } from '../../stores/localization';
import type { ExtensionInstallation } from '../extension-sdk/extension-sdk.service';

export function initialiseStores(config: ExtensionInstallation): void {
  gaViewId.set(config?.googleAnalyticsViewId || '');
  gaClientId.set(config?.googleAnalyticsClientId || '');
  gaClientEmail.set(config?.googleAnalyticsClientEmail);
  gaApiKey.set(config?.googleAnalyticsKey || '');

  contentItemIdMapping.set(config?.mappings?.contentItemId);
  editionIdMapping.set(config?.mappings?.editionId);
  slotIdMapping.set(config?.mappings?.slotId);

  contentItemFilter.set(config?.filters?.contentItemFilter);
  editionFilter.set(config?.filters?.editionFilter);
  slotFilter.set(config?.filters?.slotFilter);

  breakdown.set(
    config?.breakdown || {
      dimension: 'ga:deviceCategory',
      title: 'Device Breakdown',
    }
  );

  if (config?.localization?.locale) {
    locale.set(config?.localization?.locale);
  }

  if (config?.localization?.currencyCode) {
    currencyCode.set(config?.localization?.currencyCode);
  }
}
