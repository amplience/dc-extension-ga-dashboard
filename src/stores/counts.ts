import { writable, derived } from 'svelte/store';
import type {
  SearchIndexUsersCount,
  SearchIndexSearchesCount,
  SearchIndexNoResultsRate,
} from 'dc-management-sdk-js';

export interface Counts {
  usersCount?: SearchIndexUsersCount;
  searchesCount?: SearchIndexSearchesCount;
  noResultsRate?: SearchIndexNoResultsRate;
}

export const counts = writable<Counts>({});

export const searchesPerUser = derived(counts, ($counts) => {
  return $counts?.usersCount?.count && $counts?.searchesCount?.count
    ? Math.round(
        ($counts?.searchesCount?.count / $counts?.usersCount?.count) * 1e1
      ) / 1e1
    : 0;
});
