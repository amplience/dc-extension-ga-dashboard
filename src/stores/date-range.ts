import { writable } from './persisted-store';
import { formatDateAsISOString } from '../utils/date-format';

export const DAY = 24 * 60 * 60 * 1000;

export interface DateRange {
  from: string;
  to: string;
}

const NOW = new Date().setHours(0, 0, 0, 0).valueOf();

export const dateRange = writable<DateRange>('date-range', {
  from: formatDateAsISOString(new Date(NOW - DAY * 30)),
  to: formatDateAsISOString(new Date(NOW - DAY)),
});
