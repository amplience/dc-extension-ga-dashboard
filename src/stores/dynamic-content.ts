import type { DynamicContent, Hub } from 'dc-management-sdk-js';
import { writable } from 'svelte/store';

export const client = writable<DynamicContent>(null);

export const hub = writable<Hub>(null);
