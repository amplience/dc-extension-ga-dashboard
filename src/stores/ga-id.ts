import { writable as persistedWritable } from './persisted-store';

export const gaId = persistedWritable<string>('ga-id', null);
