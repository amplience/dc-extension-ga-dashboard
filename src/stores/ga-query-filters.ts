import { derived } from 'svelte/store';
import { editionIdMapping } from './google-analytics';
import { selectedEdition } from './selected-edition';
import { writable } from 'svelte/store';

export const contentItemFilter = writable<string>(null);
export const editionFilter = writable<string>(null);
export const slotFilter = writable<string>(null);

export const gaQueryFilter = derived(
  [selectedEdition, editionIdMapping],
  ([$selectedEdition, $editionIdMapping]) => {
    if ($selectedEdition) {
      return $editionIdMapping + '==' + $selectedEdition.id;
    }
    return undefined;
  }
);

export function constructFilter(
  queryFilter: string,
  customFilter: string
): string {
  if (customFilter) {
    return queryFilter ? `${queryFilter};${customFilter}` : customFilter;
  }

  return queryFilter;
}
