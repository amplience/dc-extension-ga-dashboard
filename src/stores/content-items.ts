import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createContentItems() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedContentItem = get(contentItems).find(
        (item) => item.id === id
      );
      if (cachedContentItem) {
        return cachedContentItem;
      }
      const contentItem = await get(client).contentItems.get(id);
      update((n) => [...n, ...[contentItem]]);
      return contentItem;
    },
    reset: () => set([]),
  };
}

export const contentItems = createContentItems();
