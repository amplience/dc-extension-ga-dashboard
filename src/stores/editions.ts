import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createEditions() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedEdition = get(editions).find((item) => item.id === id);
      if (cachedEdition) {
        return cachedEdition;
      }
      const edition = await get(client).editions.get(id);
      update((n) => [...n, ...[edition]]);
      return edition;
    },
    reset: () => set([]),
  };
}

export const editions = createEditions();
