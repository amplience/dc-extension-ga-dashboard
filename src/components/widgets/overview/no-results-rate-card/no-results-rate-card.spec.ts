import { render } from '@testing-library/svelte';
import NoResultsRateCard from './no-results-rate-card.svelte';
import { counts, Counts } from '../../../../stores/counts';

describe('NoResultsRateCard', () => {
  it('should render the NoResultsRateCard component with normal text hint', () => {
    counts.set(({
      noResultsRate: { rate: 0.12 },
    } as unknown) as Counts);
    const { container } = render(NoResultsRateCard, {});

    expect(container.firstChild).toMatchSnapshot();
  });
});
