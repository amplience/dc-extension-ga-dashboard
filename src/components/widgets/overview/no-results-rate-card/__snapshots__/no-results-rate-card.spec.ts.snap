// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResultsRateCard should render the NoResultsRateCard component with normal text hint 1`] = `
<div>
  <article
    class="svelte-14k6rag"
    style="--overview-card-color: #cd69a3;"
  >
    <div
      class="overview-card-top svelte-14k6rag"
    >
      <div
        class="overview-card-heading svelte-14k6rag"
      >
        <div
          class="overview-card-title svelte-14k6rag"
        >
          <div
            class=" svelte-zpbvvv"
            style="--icon-width: 24px; --icon-height: 24px;"
          >
            <object
              src="ic-asset-no-results.svg/"
            />
          </div>
           
          <h2
            class="svelte-14k6rag"
          >
            No results rate
          </h2>
        </div>
         
        <div
          class="overview-card-value svelte-14k6rag"
        >
          12.00%
        </div>
      </div>
       
      <div
        class="overview-card-content svelte-14k6rag"
      >
        <div
          slot="body"
        >
          Percentage of searches for which no results were found
        </div>
      </div>
    </div>
     
    <div
      class="overview-card-footer svelte-14k6rag"
    />
  </article>
</div>
`;
